File: app/src/main/java/net/ginteam/carmen/contract/category/CategoriesContract.java
package net.ginteam.carmen.contract.category;

import net.ginteam.carmen.contract.FetchContract;
import net.ginteam.carmen.model.category.CategoryModel;

import java.util.List;

/**
 * Created by Eugene on 12/23/16.
 */

public interface CategoriesContract {

    interface View extends FetchContract.View {

        void showCategories(List <CategoryModel> categories);

    }

    interface Presenter extends FetchContract.Presenter <View> {

        void selectCategory(CategoryModel category);

    }

}

File: app/src/main/kotlin/net/ginteam/carmen/kotlin/contract/CategoriesContract.kt
package net.ginteam.carmen.kotlin.contract

import net.ginteam.carmen.kotlin.model.CategoryModel

/**
 * Created by eugene_shcherbinock on 2/13/17.
 */
object CategoriesContract {

    interface View : BaseContract.View {

        fun showCategories(categories: List <CategoryModel>)

    }

    interface Presenter : BaseContract.Presenter<View> {

        fun fetchCategories()

    }

}